cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -889,26 +890,28 @@ def _add_dynamo_db_table(self):
         self._dynamo_db_table = table
 
     def _add_head_node(self):
+        head_node = self._cluster_config.head_node
```

Comment:
for the _cluster_config field, I am concern about we store config object both in Cluster and ClusterCdkStack as an instance field. But Cluster also has ClusterCdkStack instance as a field. Is it ok for this dependency relationship? Or we should pass config to these private functions rather than store config as an instance field.

Reply:
`Cluster` doesn't have `ClusterCdkStack` instance as a field.
`Cluster` just calls `CDKTemplateBuilder().build_cluster_template()` static method.

`_cluster_config` is just an internal reference to the the configuration we're passing to the ClusterCdkStack.
Let's talk about it.